iwyu_globals.cc

```diff
@@ -117,6 +117,9 @@ static void PrintHelp(const char* extra_msg) {
          "   --update_comments: update and insert 'why' comments, even if no\n"
          "        #include lines need to be added or removed.\n"
          "   --no_fwd_decls: do not use forward declarations.\n"
+         "   --no_fwd_decls_source_only: do not use forward declarations\n"
```

Comment:
I think I would prefer to make `--no_fwd_decls` take an optional argument, and:

* Add an `enum class NoFwdDecls` with enumerators `Disabled`, `Everywhere` and `SourcesOnly`
* Let it default to `Disabled`
* If  `--no_fwd_decls=everywhere` or `--no_fwd_decls` is provided without argument, set it to `Everywhere`
* If `--no_fwd_decls=src-only` is provided, set it to `SourcesOnly`
* If any other argument, fail with an error

It even feels like we could generalize `--no_fwd_decls` to some kind of `--policy_tweak`, but I'm not sure how well that scales. Small steps.

I'm not loving my name suggestions, but something to that effect would be nicer than a whole new switch.

Reply:
I've made the changes you suggested - replaced the separate boolean flags with a `NoFwdDecls` enum and made `--no_fwd_decls` accept optional arguments (`everywhere` or `src-only`). The old usage still works exactly the same way. Thanks!